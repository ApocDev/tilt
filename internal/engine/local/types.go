package local

import (
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"

	"github.com/tilt-dev/tilt/pkg/apis/core/v1alpha1"
)

type Cmd = v1alpha1.Cmd
type CmdList = v1alpha1.CmdList
type CmdStatus = v1alpha1.CmdStatus
type CmdSpec = v1alpha1.CmdSpec
type CmdStateWaiting = v1alpha1.CmdStateWaiting
type CmdStateTerminated = v1alpha1.CmdStateTerminated
type CmdStateRunning = v1alpha1.CmdStateRunning
type CmdServer = v1alpha1.CmdServer
type CmdServerSpec = v1alpha1.CmdServerSpec
type CmdServerStatus = v1alpha1.CmdServerStatus
type ObjectMeta = metav1.ObjectMeta
