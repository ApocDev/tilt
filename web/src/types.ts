export enum ResourceView {
  Log,
  Preview,
  Alerts,
}

export enum TriggerMode {
  TriggerModeAuto,
  TriggerModeManual,
}

export type Build = {
  Error: {} | string | null
  StartTime: string
  Log: string
  FinishTime: string
  Edits: Array<string> | null
  IsCrashRebuild: boolean
  Warnings: Array<string> | null
}

export type TiltBuild = {
  Version: string
  Date: string
  Dev: boolean
}

// what is the status of the resource in the cluster
export enum RuntimeStatus {
  Ok = "ok",
  Pending = "pending",
  Error = "error",
}

// What is the status of the resource with respect to Tilt
export enum ResourceStatus {
  BuildQueued, // in auto, if you have changed a file but an affected build hasn't started yet. In manual after you have clicked build, before it has started building
  Building,
  Error,
  Warning,
  Deploying,
  Deployed, // defer to RuntimeStatus
}

export type Resource = {
  Name: string
  CombinedLog: string
  BuildHistory: Array<any>
  CrashLog: string
  CurrentBuild: any
  DirectoriesWatched: Array<any>
  Endpoints: Array<string>
  PodID: string
  IsTiltfile: boolean
  LastDeployTime: string
  PathsWatched: Array<string>
  PendingBuildEdits: Array<string>
  PendingBuildReason: number
  PendingBuildSince: string
  ResourceInfo: {
    PodCreationTime: string
    PodLog: string
    PodName: string
    PodRestarts: number
    PodUpdateStartTime: string
    YAML: string
    PodStatus: string
    Endpoints: Array<string>
  }
  RuntimeStatus: string
  TriggerMode: TriggerMode
  HasPendingChanges: boolean
}
