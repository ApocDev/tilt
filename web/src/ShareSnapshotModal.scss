@import "constants";

.modal {
  position: fixed;
  top: 0;
  left: 0;
  width:100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.6);
}

.modal-main {
  position:fixed;
  background: $color-white;
  color: $color-gray-darkest;
  width: 80%;
  height: auto;
  top:50%;
  left:50%;
  transform: translate(-50%,-50%);
}

.display-block {
  display: block;
}

.display-none {
  display: none;
}

.TopBar-snapshotUrlWrap {
  flex: 1;
  display: flex;
  justify-content: flex-end;
}


.TopBar-snapshotUrlWrap button {
  background-color: transparent;
  border: 1px solid rgba($color-white, $translucent-ish);
  color: $color-white;
  text-transform: uppercase;
  padding-top: $spacing-unit / 3;
  padding-bottom: $spacing-unit / 3;
  padding-left: $spacing-unit / 2;
  padding-right: $spacing-unit / 2;
}

.TopBar-snapshotUrlWrap button:hover {
  cursor: pointer;
  background-color: $color-gray-dark;
}

.TopBar-snapshotUrl {
  background-color: $color-gray-dark;
  border: 1px solid $color-gray-darkest;
  padding-left: $spacing-unit / 2;
  padding-right: $spacing-unit / 2;
  user-select: all;
}
