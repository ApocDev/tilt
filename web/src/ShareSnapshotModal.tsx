import React, { PureComponent } from "react"
import Modal from "react-modal"
import "./ShareSnapshotModal.scss"
import cookies from "js-cookie"

type props = {
  handleSendSnapshot: () => void
  handleClose: () => void
  snapshotUrl: string
  tiltCloudUsername: string | null
  tiltCloudSchemeHost: string
  isOpen: boolean
}

export default class ShareSnapshotModal extends PureComponent<props> {
  render() {
    let tcUser = this.props.tiltCloudUsername;
    let user = tcUser ? tcUser : "[anonymous user]";
    let link = this.renderShareLink();
    return (
      <Modal
        onRequestClose={this.props.handleClose}
        isOpen={this.props.isOpen}
        className="ShareSnapshotModal"
      >
        <h2 className="ShareSnapshotModal-title">Share a Shapshot</h2>
        <div className="ShareSnapshotModal-pane">
          <p>Let anyone explore your current Tilt session with an interactive snapshot.</p>
          {link}
          <p className="ShareSnapshotModal-user">Sharing as <strong>{user}</strong></p>
        </div>
        <div className="ShareSnapshotModal-pane">
          {this.tiltCloudCopy()}
        </div>
      </Modal>
    )
  }

  renderShareLink() {
    const hasLink = this.props.snapshotUrl !== "";
    if (!hasLink) {
      return (
        <button onClick={this.props.handleSendSnapshot}>Share Snapshot</button>
      )
    }
    return (
      <a href={this.props.snapshotUrl} target="_blank">
        Open link in new tab
      </a>
    )
  }

  tiltCloudCopy() {
    if (!this.props.tiltCloudUsername) {
      return (
        <section className="ShareSnapshotModal-tiltCloudCopy">
          <p>Register on TiltCloud to share under your name, view, and delete snapshots.</p>
          <form
            className="ShareSnapshotModal-tiltCloudButtonForm"
            action={this.props.tiltCloudSchemeHost + "/register_token"}
            target="_blank"
            method="POST"
            onSubmit={ShareSnapshotModal.notifyTiltOfRegistration}
          >
            <p>You'll need a GitHub account</p>
            <span className="ShareSnapshotModal-tiltCloudButtonForm-spacer" />
            <input className="ShareSnapshotModal-tiltCloudButton" type="submit" value="Sign Up" />
            <input
              name="token"
              type="hidden"
              value={cookies.get("Tilt-Token")}
            />
          </form>
        </section>
      )
    } else {
      return (
        <p>View and delete snapshots on <a href="http://cloud.tilt.dev/snapshots" target="_blank">TiltCloud</a></p>
      )
    }
  }

  static notifyTiltOfRegistration() {
    let url = `${window.location.protocol}//${
      window.location.host
    }/api/user_started_tilt_cloud_registration`
    fetch(url, {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
    })
  }
}
