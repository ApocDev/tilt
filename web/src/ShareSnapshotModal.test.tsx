import React from "react"
import renderer from "react-test-renderer"
import ShareSnapshotModal from "./ShareSnapshotModal"
import { Snapshot } from "./types"

const testState: Snapshot = {
  Message: "",
  View: null,
  IsSidebarClosed: false,
}
const fakeSendsnapshot = (snapshot: Snapshot) => void {}
const fakeHandleCloseModal = () => {}

describe("ShareSnapshotModal", () => {
  it("renders with modal open", () => {
    const tree = renderer
      .create(
        <ShareSnapshotModal
          handleSendSnapshot={fakeSendsnapshot}
          handleClose={fakeHandleCloseModal}
          show={true}
          state={testState}
          snapshotUrl="http://test.com"
          registerTokenUrl="http://registertoken.com"
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("renders with modal closed", () => {
    const tree = renderer
      .create(
        <ShareSnapshotModal
          handleSendSnapshot={fakeSendsnapshot}
          handleClose={fakeHandleCloseModal}
          show={false}
          state={testState}
          snapshotUrl="http://test.com"
          registerTokenUrl="http://registertoken.com"
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
