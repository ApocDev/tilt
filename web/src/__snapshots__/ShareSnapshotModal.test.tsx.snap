// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareSnapshotModal renders with modal closed 1`] = `
<div
  className="modal display-none"
>
  <section
    className="modal-main"
  >
    <section
      className="modal-snapshotUrlWrap"
    >
      <button
        onClick={[Function]}
      >
        close
      </button>
      <button
        onClick={[Function]}
      >
        Open link in new tab
      </button>
    </section>
    <hr />
    <section
      className="modal-cloud"
    >
      <p>
        Go to 
        <a
          href=""
        >
          TiltCloud
        </a>
         to manage your snapshots.
        <br />
        (You'll just need to link your GitHub Account)
      </p>
    </section>
  </section>
</div>
`;

exports[`ShareSnapshotModal renders with modal open 1`] = `
<div
  className="modal display-block"
>
  <section
    className="modal-main"
  >
    <section
      className="modal-snapshotUrlWrap"
    >
      <button
        onClick={[Function]}
      >
        close
      </button>
      <button
        onClick={[Function]}
      >
        Open link in new tab
      </button>
    </section>
    <hr />
    <section
      className="modal-cloud"
    >
      <p>
        Go to 
        <a
          href=""
        >
          TiltCloud
        </a>
         to manage your snapshots.
        <br />
        (You'll just need to link your GitHub Account)
      </p>
    </section>
  </section>
</div>
`;
